diff -ruN /nix/store/aig3ajvcpp7j89zzj81cym15kxydgaa4-tinygo-0.40.1/share/tinygo/src/sync/waitgroup.go /tmp/full_tgo/src/sync/waitgroup.go
--- /nix/store/aig3ajvcpp7j89zzj81cym15kxydgaa4-tinygo-0.40.1/share/tinygo/src/sync/waitgroup.go	1969-12-31 19:00:01.000000000 -0500
+++ /tmp/full_tgo/src/sync/waitgroup.go	2026-06-28 20:22:10.768924462 -0400
@@ -83,3 +83,15 @@
 		}
 	}
 }
+
+// Go calls the given function in a new goroutine.
+// It blocks until the goroutine has started.
+// It adds 1 to the WaitGroup counter before starting, and
+// calls Done when the goroutine completes.
+func (wg *WaitGroup) Go(f func()) {
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		f()
+	}()
+}
