--- a/src/os/file_unix_chown.go
+++ b/src/os/file_unix_chown.go
@@ -0,0 +1,23 @@
+//go:build darwin || (linux && !baremetal && !wasm_unknown && !nintendoswitch) || wasip1 || wasip2
+
+package os
+
+import "syscall"
+
+// Chown changes the owner and group ids of the named file.
+func (f *File) Chown(uid, gid int) error {
+	if f == nil {
+		return ErrInvalid
+	}
+	if f.handle == nil {
+		return ErrClosed
+	}
+	longName := fixLongPath(f.name)
+	e := ignoringEINTR(func() error {
+		return syscall.Chown(longName, uid, gid)
+	})
+	if e != nil {
+		return &PathError{Op: "chown", Path: f.name, Err: e}
+	}
+	return nil
+}
