--- a/src/net/tcpsock.go
+++ b/src/net/tcpsock.go
@@ -12,6 +12,7 @@
 	"internal/itoa"
 	"io"
 	"net/netip"
+	"os"
 	"strconv"
 	"syscall"
 	"time"
@@ -282,6 +283,24 @@
 	return fmt.Errorf("CloseWrite not implemented")
 }
 
+// File returns a copy of the underlying file descriptor.
+func (c *TCPConn) File() (*os.File, error) {
+	return nil, errors.New("TCPConn.File not implemented")
+}
+
+// KeepAliveConfig contains TCP keep-alive options.
+type KeepAliveConfig struct {
+	Enable   bool
+	Idle     time.Duration
+	Interval time.Duration
+	Count    int
+}
+
+// SetKeepAliveConfig configures keep-alive messages sent by the operating system.
+func (c *TCPConn) SetKeepAliveConfig(config KeepAliveConfig) error {
+	return errors.New("TCPConn.SetKeepAliveConfig not implemented")
+}
+
 type listener struct {
 	fd    int
 	laddr *TCPAddr
